src/app/shared/components/addscenario/addscenario.component.ts

```diff
@@ -152,6 +162,82 @@ export class AddScenarioModal implements OnInit, OnDestroy {
                 this.cancelCreateScenario();
             }
         );
+
+        if (this.clone == true){
+            this.clearScenario();
+            this.cloneSenario();
+        }  
+    }
+
+    cloneSenario(){  
```

Comment:
minor typo here, "Scenario", not "Senario", make sure it's updated anywhere it calls it (e.g. line 168). Using that Code Spell Checker extension in VScode might catch that.